File: tests_wraps/src/main/java/com/github/klee0kai/test/car/di/bestOf/red/RedCarComponent.java
package com.github.klee0kai.test.car.di.bestOf.red;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.test.car.di.bestOf.both.BothCarModule;
import com.github.klee0kai.test.car.model.Car;

import javax.inject.Named;
import java.lang.ref.WeakReference;

@Component
public interface RedCarComponent {

    BothCarModule myModule();

    @Named("redCar")
    WeakReference<Car> redCar();

}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/di/bestOf/red/RedCarComponent.kt
package com.github.klee0kai.test.car.di.bestOf.red

import com.github.klee0kai.stone.annotations.component.*
import com.github.klee0kai.test.car.di.bestOf.both.BothCarModule
import com.github.klee0kai.test.car.model.Car
import java.lang.ref.WeakReference
import javax.inject.Named

@Component
interface RedCarComponent {
    fun myModule(): BothCarModule

    @Named("redCar")
    fun redCar(): WeakReference<Car?>
}
